src/data_snack/snack.py

```diff
@@ -13,66 +14,113 @@ def _build_key(type_name: Text, *keys: Text):
 
 @dataclass
 class Snack:
+    """
+    A core interface handling saving and reading Entities from db.
+    """
+
     connection: Connection
     registry: Dict[Text, EntityRegistry] = field(default_factory=dict)
 
-    def register_entity(self, cls: EntityType, keys: List[Text], serializer: Serializer = None):
+    def register_entity(self, cls: Type[Entity], key_fields: List[Text], serializer: Serializer = None) -> None:
+        """
+        Registers new Entity type to Snack.
+
+        :param cls: Entity type
+        :param key_fields: a list of fields that will be used to define Entity key
+        :param serializer: Serializer that is used to compress and decompress entities before saving in db
+        """
         type_name = cls.__name__
 
         if type_name in self.registry:
-            raise EntityAlreadyRegistered(f"Entity {type_name} is already registred")
+            raise EntityAlreadyRegistered(f"Entity {type_name} is already registered")
 
-        if not keys:
+        if not key_fields:
```

Comment:
Is this needed? `key_fields` has no default, so the error would be raised if not passed.

Reply:
you can pass an empty list here